scylla/src/network/connection_pool.rs

```diff
@@ -226,6 +226,17 @@ impl NodeConnectionPool {
         }
     }
 
+    pub(crate) fn is_connected(&self) -> bool {
+        let maybe_conns = self.conns.load();
+        match maybe_conns.as_ref() {
+            MaybePoolConnections::Initializing => false,
+            MaybePoolConnections::Broken(_) => false,
+            // TODO: Is is possible for the pool to be "Ready", but for "_pool_connections"
+            // to be empty?
+            MaybePoolConnections::Ready(_pool_connections) => true,
```

Comment:
:question: Is it?

Reply:
```rust
enum MaybePoolConnections {
    // The pool is being filled for the first time
    Initializing,

    // The pool is empty because either initial filling failed or all connections
    // became broken; will be asynchronously refilled. Contains an error
    // from the last connection attempt.
    Broken(ConnectionError),

    // The pool has some connections which are usable (or will be removed soon)
    Ready(PoolConnections),
}
```
From this enum I understand that if it's not buggy, then `Ready` implies some connections in the pool.

Reply:
Edited the comment.